crates/oxc_linter/src/rules/unicorn/no_zero_fractions.rs

```diff
@@ -146,22 +177,28 @@ fn test() {
         (r"const foo = 1.", r"const foo = 1"),
         (r"const foo = +1.", r"const foo = +1"),
         (r"const foo = -1.", r"const foo = -1"),
-        // maybe todo
-        // In the following tests, the comments did not pass the fixer.
-
-        // (r"const foo = 1.e10", r"const foo = 1e10"),
-        // (r"const foo = +1.e-10", r"const foo = +1e-10"),
-        // (r"const foo = -1.e+10", r"const foo = -1e+10"),
+        (r"const foo = 1.e10", r"const foo = 1e10"),
+        (r"const foo = +1.e-10", r"const foo = +1e-10"),
+        (r"const foo = -1.e+10", r"const foo = -1e+10"),
         (r"const foo = (1.).toString()", r"const foo = (1).toString()"),
-        // (r"1.00.toFixed(2)", r"(1).toFixed(2)"),
-        // (r"1.00 .toFixed(2)", r"(1) .toFixed(2)"),
+        (r"1.00.toFixed(2)", r"(1).toFixed(2)"),
```

Comment:
Could this break ASI?
```js
console.log('foo')
1.00.toFixed(2) // -> (1).toFixed(2), parsed as a function call
```

Reply:
i will fix it

Reply:
I need help, I don't know how to get the previous token.

Reply:
this case is also a problem blocking #4312. I'll add a task to add a `delete_statement` method to `RuleFixer`.

Reply:
Sorry, I don't know what `delete_statement` does? The logic I need to implement here is It : 

checks the type and value of tokenBefore:
If tokenBefore is a Punctuator (e.g., a symbol like ;, ], or )), it determines whether a semicolon is necessary based on the context (e.g., the type of the last block node).
If the token type is in tokenTypesNeedsSemicolon, it returns true (semicolon needed).
Special cases like Template strings, ObjectExpression blocks, and certain Identifier cases are handled explicitly. 

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/5c92a9aed7793afa00f23e72e25bc58c10adecd4/rules/utils/needs-semicolon.js#L33

Is this also a blocking problem or is there an API available?

Reply:
This API does not exist yet, we should move ahead with this improvement for the time being.

Reply:
> This API does not exist yet, we should move ahead with this improvement for the time being.

Since it is not a problem with the current rules, let’s merge this PR first. Then join a new issue to follow up on this API